code/core/src/preview-api/modules/store/csf/prepareStory.ts

```diff
@@ -227,6 +233,7 @@ function preparePartialAnnotations<TRenderer extends Renderer>(
     parameters,
     initialArgs: passedArgs,
     argTypes: passedArgTypes,
+    storyGlobals,
```

Comment:
I would just call this `globals`.

Reply:
> The divergence in types between PreparedStory and StoryContext definitely has the potential to be confusing.

IMO, they should diverge. I think with CSF4 we should call them just `Story` and `StoryContext` and `Story` is what we return from the csf factories:

```tsx
const Basic = defineStory({ // <-- name up to debate
  args: { ... }
})
```

And `StoryContext` is what we provide as argument to play, render, decorators etc. and it has 
* extra stuff like `canvasElement` and `abortSignal`
* will resolve the `context` argument for "properties/fixtures" like `mount`, `renderToCanvas`, `step`
* integration such as storybook can override values such as `args` and `globals` in their UI

> Maybe we shouldn't make PreparedStory extend StoryContext?
Probably PreparedStory should extend StoryAnnotations and ProjectAnnotations? It is kind of the resolved version of that.

In some sense, the relationship is kind of the opposite at the moment at runtime, StoryContext extends PrepareStory. But because of that, this stuff:
https://github.com/storybookjs/storybook/blob/c11a299905f3fd3e77f63552e2bcde2b2180e4fe/code/lib/types/src/modules/story.ts#L100-L108

Actually sneakily ends up on the story context on runtime
![image](https://github.com/user-attachments/assets/04bda0d1-4263-4c44-b88b-f40f42ed31f7)

We should probably get rid of that?

Reply:
Agree with all of that.

Reply:
@kasperpeulen what's the action that follows out of this discussion that needs to happen to this PR?